src/pages/academy/search/index.tsx

```diff
@@ -312,6 +312,106 @@ const SearchPage = () => {
+
+                {article_result.length !== 0 ? (
+                    getPaginatedArticles()
+                ) : (
+                    <Flex m="16px 0">
+                        <Header as="h3" type="subtitle-2" weight="normal" color="grey-5">
+                            No results found
+                        </Header>
+                    </Flex>
+                )}
+
+                {full_article_link && !items_type && total_article > 4 && (
+                    <AllArticleButton tertiary="true" to={full_article_link}>
+                        All article results
+                    </AllArticleButton>
+                )}
+            </>
+        )
+    }
+
+    const renderVideo = () => {
+        return (
+            <>
+                <StyledHeaderWrapper jc="space-between">
+                    <Header as="h3" type="subtitle-2">
+                        Videos
+                    </Header>
+                    <Header as="span" type="paragraph-2" align="right">
+                        <>
+                            {total_video > 1 && !items_type
+                                ? `1-2 of ${total_video} results`
+                                : `${total_video} results`}
+                        </>
+                    </Header>
+                </StyledHeaderWrapper>
+
+                {video_result.length != 0 ? (
+                    <>
+                        <VideoParentWrapper
+                            closeVideo={closeVideo}
+                            currentVideoItems={video_result}
+                            openVideo={openVideo}
+                            show={show}
+                            video_src={play_video_src}
+                        />
+                    </>
+                ) : (
+                    <Flex m="16px 0">
+                        <Header as="h3" type="subtitle-2" weight="normal" color="grey-5">
+                            No results found
+                        </Header>
+                    </Flex>
+                )}
+
+                {full_video_link && !items_type && total_video > 1 && (
+                    <AllArticleButton tertiary="true" to={full_video_link}>
+                        All video results
+                    </AllArticleButton>
+                )}
+            </>
+        )
+    }
+    const checkFilterType = (url, is_search = false) => {
+        const type = queryParams.get('type')
+        const category = queryParams.get('category')
+        if (!is_search) {
+            return type === url
+        } else {
+            return category && type ? false : true
+        }
+    }
+
+    const isShowArticle = () => {
+        const is_article = checkFilterType('article')
+        const is_search = checkFilterType('', true)
+        const is_video = checkFilterType('video')
+        if (is_article || is_search) {
+            return true
+        }
+        if (is_video) {
+            return false
+        }
+    }
```

Comment:
We can remove the `checkFilterType` function and do the comparison directly in `isShowArticle` function

Reply:
okay